include/verilated_threads.cpp

```diff
@@ -82,143 +81,42 @@ void VlWorkerThread::workerLoop() {
             work.m_fnp = nullptr;
         }
     }
-
-    if (VL_UNLIKELY(m_profiling)) m_poolp->tearDownProfilingClientThread();
 }
 
-void VlWorkerThread::startWorker(VlWorkerThread* workerp) {
+void VlWorkerThread::startWorker(VlWorkerThread* workerp, uint32_t threadId,
+                                 VlExecutionProfiler* profilerp) {
     Verilated::threadContextp(workerp->m_contextp);
+#ifdef VL_PROFILER
+    // Note: setupThread is not defined without VL_PROFILER, hence the #ifdef. Still, we might
+    // not be profiling execution (e.g.: PGO only), so profilerp might still be nullptr.
+    if (profilerp) profilerp->setupThread(threadId);
```

Comment:
```suggestion
    if (VL_UNLIKELY(profilerp)) profilerp->setupThread(threadId);
```

Reply:
This is in a constructor, so executed once through the lifetime of the model. I think overusing branch hints where not important just makes things more verbose/less readable.